src/morse/index.js

```diff
@@ -58,6 +58,51 @@ const MORSE_CODE = {
 
 Object.freeze(MORSE_CODE);
 
-function morse(text) {}
+function morse(text) {
+
+  
+  /* 
+    Steps to appraoching Morse code translation to readable words
+    1. Split morse text in words using the split method
+    2. Loop through splitted morse, pushing into a morseToWordArray
+    3. replace undefined and null embedded with any word
+  */
+
+
+  try{
+    if(typeof(text)!=='string'){
+      throw "Please provide a morse string"
+    }
+    if(text==='undefined'){
+      throw ""
+    }
+    if(text===""){
+      return ""
+    }
+
+    let splittedMorse = text.split(' ')
+    let morseToWordArray = []
+
+    for (let i = 0; i < splittedMorse.length; i++) {
+      morseToWordArray.push(MORSE_CODE[splittedMorse[i]])
+    }
+
+    let combinedLetters = morseToWordArray.reduce((accumulator, current) => accumulator + current)
+    let noNullUndefinedInWord = combinedLetters.replace(/undefined|NAN/gi, " ")
```

Comment:
The design that lead to this should be fixed. This could for example be fixed by the `reduce`, its not only useful for adding things. Or, you could for example run a `filter` before it.